From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct wake_irq
Patch-mainline: never, kabi
References: kabi

In 4.4.42, commit b0cb7f60998c776fa5696f24a21b685954b15881 (PM /
wakeirq: Fix dedicated wakeirq for drivers not using autosuspend),
upstream commit bed570307ed78f21b77cb04a1df781dee4a8f05a
changed some members of struct wake_irq and kABI checker started to
report breakages.

The structure is internal-only, so protect the change by __GENKSYMS__
and be done with it.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/base/power/power.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/base/power/power.h
+++ b/drivers/base/power/power.h
@@ -27,8 +27,13 @@ extern void pm_runtime_remove(struct dev
 
 struct wake_irq {
 	struct device *dev;
+#ifndef __GENKSYMS__
 	unsigned int status;
+#endif
 	int irq;
+#ifdef __GENKSYMS__
+	bool dedicated_irq:1;
+#endif
 };
 
 extern void dev_pm_arm_wake_irq(struct wake_irq *wirq);
